pkg/operator/hive/hiveadmission.go

```diff
@@ -56,33 +68,139 @@ func (r *ReconcileHiveConfig) deployHiveAdmission(hLog log.FieldLogger, h *resou
+	dnsZonesWebhook := util.ReadValidatingWebhookConfigurationV1Beta1OrDie(asset, scheme.Scheme)
+
+	// If on 3.11 we need to set the service CA on the apiservice.
+	is311, err := r.is311(hLog)
+	if err != nil {
+		hLog.Error("error detecting 3.11 cluster")
+		return err
 	}
-	for _, a := range applyAssets {
-		err = util.ApplyAsset(h, a, hLog)
+	if is311 {
+		hLog.Debug("3.11 cluster detected, modifying objects for CA certs")
+		err = r.injectCerts(apiService, []*admregv1.ValidatingWebhookConfiguration{cdWebhook, cisWebhook, dnsZonesWebhook}, hLog)
 		if err != nil {
+			hLog.WithError(err).Error("error injecting certs")
 			return err
 		}
 	}
 
+	err = h.ApplyRuntimeObject(apiService, s)
+	if err != nil {
+		hLog.WithError(err).Error("error applying apiservice")
+		return err
+	}
+	hLog.Info("apiservice applied")
+
+	err = h.ApplyRuntimeObject(cdWebhook, s)
+	if err != nil {
+		hLog.WithError(err).Error("error applying cluster deployment webhook")
+		return err
+	}
+	hLog.Info("cluster deployment webhook applied")
+
+	err = h.ApplyRuntimeObject(cisWebhook, s)
+	if err != nil {
+		hLog.WithError(err).Error("error applying cluster image set webhook")
+		return err
+	}
+	hLog.Info("cluster image set webhook applied")
+
+	err = h.ApplyRuntimeObject(dnsZonesWebhook, s)
+	if err != nil {
+		hLog.WithError(err).Error("error applying dns zones webhook")
+		return err
+	}
+	hLog.Info("dns zones webhook applied")
+
 	hLog.Info("hiveadmission components reconciled successfully")
 
 	return nil
 }
+
+func (r *ReconcileHiveConfig) injectCerts(apiService *apiregistrationv1.APIService, webhooks []*admregv1.ValidatingWebhookConfiguration, hLog log.FieldLogger) error {
+
+	// Locate the kube CA by looking up secrets in kube-system namespace, finding one of
+	// type 'kubernetes.io/service-account-token', and reading the CA off it.
+	hLog.Debug("listing secrets in hive namespace")
+	secrets := &corev1.SecretList{}
+	err := r.globalClient.List(context.Background(), &client.ListOptions{Namespace: hiveNamespace}, secrets)
+	if err != nil {
+		hLog.WithError(err).Error("error listing secrets in hive namespace")
+		return err
+	}
+	var firstSATokenSecret *corev1.Secret
+	hLog.Debugf("found %d secrets", len(secrets.Items))
+	for _, s := range secrets.Items {
+		if s.Type == corev1.SecretTypeServiceAccountToken {
+			firstSATokenSecret = &s
+			break
+		}
+	}
+	if firstSATokenSecret == nil {
+		return fmt.Errorf("no %s secrets found", corev1.SecretTypeServiceAccountToken)
+	}
+	kubeCA, ok := firstSATokenSecret.Data["ca.crt"]
+	if !ok {
+		return fmt.Errorf("secret %s did not contain key ca.crt", firstSATokenSecret.Name)
+	}
+	hLog.Debugf("found kube CA: %s", string(kubeCA))
+
+	// Load the service CA:
+	svcCertSignerSecret := &corev1.Secret{}
+	err = r.globalClient.Get(context.Background(), types.NamespacedName{Namespace: "openshift-service-cert-signer", Name: "service-serving-cert-signer-signing-key"}, svcCertSignerSecret)
```

Comment:
@dgoodwin you should be able to get this from the same secret as above, key `service-ca.crt`
If you do, then you can get rid of the globalClient and the need for hive to access additional namespaces.

Reply:
/cc @twiest

Reply:
They don't appear to be the same value.

```
(dgoodwin@wrx {metrics} ~/go/src/github.com/openshift/hive) $ k get secret deployer-token-thhp9 -o json | jq '.data."service-ca.crt"' | sha256sum                         (hive/192-168-42-204:8443/developer/hive)
bf4c4bb9a4d807a24a397eb3612ea7e604f6d35f28fe4f77b99c3c6e6984e5df  -
(dgoodwin@wrx {metrics} ~/go/src/github.com/openshift/hive) $ k get secret -n openshift-service-cert-signer service-serving-cert-signer-signing-key -o json | jq '.data."tls.key"' | sha256sum
d316aa93b245ce45b179629f3a85ef63a960996e0ec4f8a83160b880dd862eb4  -
```

Reply:
they aren't, but I don't believe you need the signer CA, the service-ca.crt that gets mounted in the pod should be sufficient.

Reply:
This is my understanding as well.

Reply:
Updated, separate commit so we can roll back if needed.